helm-charts/cs-k8s-protection-agent/templates/docker-secret.yaml

```diff
@@ -15,7 +15,7 @@ data:
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ include "cs-k8s-protection-agent.fullname" . }}-regsecret
```

Comment:
Looking at the code, the name shouldn't change. What needs to change is the `docker-secret` yaml, and the deployment yaml should be condensed since they basically do the same thing. I can do it in another PR if you don't have the time to make the changes.

Reply:
I've condensed the secret (to use a condition) depending the value configured, and on deployment side I've used the same checks as Falcon-sensor and falcon-kac to fail the template if `crowdstrikeConfig.dockerAPIToken` is used with `image.registryConfigJSON`